example/testapp/apitest/features/support/feature_hooks.js

```diff
@@ -2,14 +2,32 @@
 var cucumberApi = require('cucumber-api');
 var Promise = require('bluebird');
 var childProcess = Promise.promisifyAll(require('child_process'), {multiArgs: true});
+var portUsed = require('tcp-port-used');
+
+waitForStartup = function (done) {
+  portUsed.check(8080, "localhost")
+    .then(function(inUse) {
+      if(inUse) {
+        done();
+      }
+      else {
+        setTimeout(function() {
+          waitForStartup(done);
+        }, 1000);
+      }
+    }, function (err) {
+      console.log("Cannot connect to server. Please make sure that your ports are configured properly");
+    });
```

Comment:
Nice recursive function.  That's the best way to handle JS polling

Here's some JS idiom tips.
```
waitForStartup = function (done) {
  portUsed.check(8080, "localhost")
  .then(function(inUse) {   // no indent for ".then"
    if (inUse) return done();  // one-liner guard
    setTimeout(function () {
      waitForStartup(done);
    }, 1000);
  .catch( function (err) {  // Use catch instead of second argument in "then". Mimics try...catch
    console.log("Cannot connect to server. Please make sure that your ports are configured properly");
    done(err);  // pass the error to "done" so that it will propagate into the BeforeFeatures error handler.
  });
}
```